.github/workflows/ci.yml

```diff
@@ -1,68 +1,182 @@
 
-    - name: Setup dotnet
-      uses: actions/setup-dotnet@v4
+  lint-md:
+    needs: detect-changes
+    if: contains(needs.detect-changes.outputs.changes, 'md')
+    uses: ./.github/workflows/markdownlint.yml
 
-    - name: Install dependencies
-      run: dotnet restore
+  lint-dotnet-format:
+    needs: detect-changes
+    if: contains(needs.detect-changes.outputs.changes, 'code')
+    uses: ./.github/workflows/dotnet-format.yml
 
-    - name: Build
-      run: dotnet build --configuration Release --no-restore --property:ExposeExperimentalFeatures=false
+  build-test-solution-stable:
+    needs: detect-changes
+    if: |
+      contains(needs.detect-changes.outputs.changes, 'code')
+      || contains(needs.detect-changes.outputs.changes, 'build')
+      || contains(needs.detect-changes.outputs.changes, 'shared')
+    uses: ./.github/workflows/Component.BuildTest.yml
+    with:
+      project-name: 'OpenTelemetry.sln'
+      project-build-commands: '-p:ExposeExperimentalFeatures=false'
+      code-cov-name: 'Solution-Stable'
 
-    - name: Test ${{ matrix.version }}
-      run: dotnet test **/bin/**/${{ matrix.version }}/*.Tests.dll --logger:"console;verbosity=detailed"
+  build-test-solution-experimental:
+    needs: detect-changes
+    if: |
+      contains(needs.detect-changes.outputs.changes, 'code')
+      || contains(needs.detect-changes.outputs.changes, 'build')
+      || contains(needs.detect-changes.outputs.changes, 'shared')
+    uses: ./.github/workflows/Component.BuildTest.yml
+    with:
+      project-name: 'OpenTelemetry.sln'
+      project-build-commands: '-p:ExposeExperimentalFeatures=true'
+      code-cov-name: 'Solution-Experimental'
 
-  build-test-experimental:
+  # Build instrumentation libraries using stable packages released to NuGet
+  build-test-instrumentation-stable:
+    needs: detect-changes
+    if: |
+      contains(needs.detect-changes.outputs.changes, 'instrumentation-packages')
+      || contains(needs.detect-changes.outputs.changes, 'build')
+      || contains(needs.detect-changes.outputs.changes, 'shared')
+    uses: ./.github/workflows/Component.BuildTest.yml
+    with:
+      project-name: './build/InstrumentationLibraries.proj'
+      project-build-commands: '-p:RunningDotNetPack=true -p:ExposeExperimentalFeatures=false'
+      code-cov-name: 'Instrumentation-Stable'
+
+  # Build instrumentation libraries using stable packages released to NuGet
+  build-test-instrumentation-experimental:
+    needs: detect-changes
+    if: |
+      contains(needs.detect-changes.outputs.changes, 'instrumentation-packages')
+      || contains(needs.detect-changes.outputs.changes, 'build')
+      || contains(needs.detect-changes.outputs.changes, 'shared')
+    uses: ./.github/workflows/Component.BuildTest.yml
+    with:
+      project-name: './build/InstrumentationLibraries.proj'
+      project-build-commands: '-p:RunningDotNetPack=true -p:ExposeExperimentalFeatures=true'
```

Comment:
For instrumentation libraries, we mainly need to test with `RunningDotNetPack` as `true` and `false` as that would decide if the instrumentation library is going to add a package reference or a project reference.

Testing with different values of `ExposeExperimentalFeatures` should only apply when we are also testing with `RunningDotNetPack` set to `false` as that would add a project reference to the API ProviderBuilderExtensions project.

Reply:
I'm not against having it but that is not important right now as we first need to first test the more important scenario of testing the instrumentation libraries against both: the latest stable API* package and the project reference of the API* package. For that, we have to test instrumentation libraries with `RunningDotNetPack` = `false` as well.

Reply:
How it is set up at the moment the project reference style (`RunningDotNetPack = false`) should be covered by the normal `OpenTelemetry.sln` builds (`build-test-solution-stable` & `build-test-solution-experimental` steps). For instrumentation-touching PRs we do an extra step which is also test them with `RunningDotNetPack = true` (`build-test-instrumentation-stable` & `build-test-instrumentation-experimental`).